File: content/problems/0819-most-common-word/solutions/normalized-word-frequency/Solution.java
class Solution {
    public String mostCommonWord(String paragraph,String[] banned){
        Set<String> blocked=new HashSet<>(Arrays.asList(banned));
        Map<String,Integer> frequency=new HashMap<>();
        String answer="";
        int best=0;
        for(String word:paragraph.toLowerCase().split("[^a-z]+")){
            if(word.isEmpty()||blocked.contains(word))continue;
            int count=frequency.merge(word,1,Integer::sum);
            if(count>best){
                best=count;
                answer=word;
            }
        }
        return answer;
    }
}

File: content/problems/0819-most-common-word/solutions/normalized-word-frequency/Solution.kt
class Solution {
    fun mostCommonWord(paragraph:String,banned:Array<String>):String{
        val blocked=banned.toSet();
        val frequency=mutableMapOf<String,Int>();
        var answer="";
        var best=0;
        for(word in paragraph.lowercase().split(Regex("[^a-z]+"))){
            if(word.isEmpty()||word in blocked)continue;
            val count=(frequency[word]?:0)+1;
            frequency[word]=count;
            if(count>best){
                best=count;
                answer=word
            }
        };
        return answer
    }
}
